### ISSUE TITLE
Support ARM64/aarch64

### ISSUE THREAD

**Issue by lareeth** [microsoft/vsts-azurevm-extension#230]
When enable Azure DevOps to use a VMSS pool configured with an ARM64 image, the deployment fails due to the following error

`The current CPU architecture is not supported. Deployment agent requires x64 architecture.`

https://github.com/microsoft/vsts-azurevm-extension/blob/f1b3ace497d1413fa0aa11d4ede646053a244cda/ExtensionHandler/Linux/src/AzureRM.py#L165-L171

What's the reason for the limitation, and are there plans to support ARM64 deployments?

I have also tried replacing the image that gets passed into the deployment

```
{
...
  "agentDownloadUrl": "https://vstsagentpackage.azureedge.net/agent/3.225.0/vsts-agent-linux-arm64-3.225.0.tar.gz",
...
}
```
But it appears the extensions throws the error before it tries to run the agent

**Comment by timarcher** [comment]
This would be extremely useful to have working for my projects as well. Use case is to build arm64 docker images that can be deployed through Azure IoT to Jetsons and raspberry pi's. Having vmss support would ease management of the build server.

**Comment by stfnzl** [comment]
Yes please, we are really looking forward for this to be fixed!

**Comment by MarianSWA** [comment]
How is this still being ignored when it's 2024 and ARM is everywhere?? We've been wanting to switch to ADO for more than 2 years now and this is the one single blocker! I expected that after all this time this would already be ready to use but every time I come back to check I am once again disappointed that it's still being ignored.

**Comment by stefanegg** [comment]
Same here; out-of-the-box support for ARM64 would be highly appreciated

**Comment by arcilli** [comment]
Any updates on this one?

**Comment by lareeth** [comment]
I would suggest emailing `opensource@microsoft.com` with something to the effect of

```
Dear Microsoft Open Source,

Please can you provide some support in prioritising a feature request for Azure DevOps, this feature request provides support for VMSS pools using ARM64 architecture. This is a important feature especially with Azure's support for ARM64 based instances, and the announcement of Server 2025 supporting ARM64.

Please can you respond to the feature request: 
Issue: https://github.com/microsoft/vsts-azurevm-extension/issues/230
Pull Request: https://github.com/microsoft/vsts-azurevm-extension/pull/232

Regards,

<Name>

**Comment by kaiyuanlim** [comment]
Hi,

Can we have this fixed?
We are moving our pipeline from Jenkins to AzureDevops and the current workaround to make this work is not that great.

**Comment by lareeth** [comment]
@LukashevichDanik @tkasparek @simonasaitta Can you explain why this issue, which has a PR to fix it already raised, is being ingored by the team? You just fixed a python issue 3 days ago.

**Comment by jgalliers** [comment]
@lareeth @kaiyuanlim @MarianSWA I have been participating in a private preview of a new Microsoft-Managed Agent Pool. I'm informed it will enter public preview in a few weeks' time (or thereabouts).

From my experiences thus far it will address most if not all of the current issues around ARM builds on Azure DevOps. Essentially you are able to specify an arbitrary VM SKU to run in your pool, along with the various scaling settings you would see. It's a blend of VMSS customisation with the Azure DevOps Agent onboard.

We have been able to successfully complete ARM-based builds and are very happy with the results. Hopefully you will soon have the ability to verify yourselves.

**Comment by kaiyuanlim** [comment]
Any updates?

**Comment by MarianSWA** [comment]
@jgalliers did they give you any updates about that new Microsoft-Managed Agent Pool?

**Comment by jgalliers** [comment]
@MarianSWA yes this turns out to be the now publicly available [Microsoft Managed a devops Pool](https://learn.microsoft.com/en-us/azure/devops/managed-devops-pools/?view=azure-devops)

It is working very well for us. It's not quite a 100% replacement yet, there's no Spot Instance option although it is coming.

**Comment by MarianSWA** [comment]
@jgalliers nice, it looks like a good start. Thanks a lot for the information.

**Comment by lukaskolafa** [comment]
@jgalliers did you manage to use the managed devops pool for arm based build agents? It looks like there is no useful image for the arm based VMs. The standard ubuntu/latest image cannot be used as it is not compatible with arm processor

**Comment by lareeth** [comment]
@lukaskolafa we have been successfull in updating our packer build agent images to build both amd64 and arm64 images, and then use those with managed devops pools without issue. The managed devops pools can use your private image gallery images instead of the azure provided images

**Comment by lukaskolafa** [comment]
> > [@lukaskolafa](https://github.com/lukaskolafa) we have been successfull in updating our packer build agent images to build both amd64 and arm64 images, and then use those with managed devops pools without issue. The managed devops pools can use your private image gallery images instead of the azure provided images
> 
> My issue with managed devops pools (I haven't tested in a month or so) is that there is ~1 minute delay from a job being ran to the agent picking up and executing the job. This is with an agent queued or not. Are you experiencing this?

We observe much longer delays, sometimes even 30 minutes. Azure docs states it can be up to 15 minutes. You can buy stand-by agents if you want to speed this up, but this is expensive.

**Comment by gavintoddchurch** [comment]
> > > [@lukaskolafa](https://github.com/lukaskolafa) we have been successfull in updating our packer build agent images to build both amd64 and arm64 images, and then use those with managed devops pools without issue. The managed devops pools can use your private image gallery images instead of the azure provided images
> > 
> > 
> > My issue with managed devops pools (I haven't tested in a month or so) is that there is ~1 minute delay from a job being ran to the agent picking up and executing the job. This is with an agent queued or not. Are you experiencing this?
> 
> We observe much longer delays, sometimes even 30 minutes. Azure docs states it can be up to 15 minutes. You can buy stand-by agents if you want to speed this up, but this is expensive.

We experience ~1 min delay even with a stand by agent
